babykicks/src/hooks/useGoogleAuth.js

```diff
@@ -0,0 +1,45 @@
+import { ProfileContext } from "../contexts/profile.context";
+import { UserContext } from "../contexts/user.context";
+import { useContext } from "react";
+import { auth, provider } from "../utils/firebase/config";
+import { signInWithPopup, signOut } from "firebase/auth";
+
+export const useGoogleAuth = () => {
+  const { setCurrentUser } = useContext(UserContext);
+  const { setCurrentProfile } = useContext(ProfileContext);
+
+  const handleSignInClick = async () => {
```

Comment:
the handleClick from the previous SignIn component was moved into the useGoogleAuth hook here. Likewise with the handleClick for the previous SignOut component.